### DISCUSSION TITLE
async version of urequests library

### DISCUSSION THREAD

**Discussion by djbloc** [micropython/micropython#12222] [Using MicroPython] [answered]
I've been using urequests successfully to download jpeg files from https websites.

Now I'm experimenting with [asyncio](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md) and I can't seem to find an async equivalent of the requests library.

I came across [uaiohttpcient ](https://github.com/micropython/micropython-lib/blob/master/micropython/uaiohttpclient/uaiohttpclient.py) in micropython-lib however that has a severe limitation that is only works for http connections.

I'm using a Pico W and would have thought this is a fairly common use case however I've been searching google for hours and nothing obvious for micropython is coming up. I'm clearly looking in the wrong places.

Please can you help by pointing to an async requests library. Many thanks

**Comment by jimmo** [comment, marked as answer]
@djbloc SSL support for asyncio is currently the last remaining item for the v1.21 release (see the recent PRs about implementing SSLContext). Once that's implemented then it will be possible to add it to the various micropython-lib libraries (like uaiohttpclient).

**Comment by djbloc** [comment]
Many thanks @jimmo

I look forward to seeing the v1.21 release
